src/query_executor.c

```diff
@@ -72,10 +72,14 @@ void ReturnClause_ExpandCollapsedNodes(RedisModuleCtx *ctx, AST_Query *ast, cons
              * Find collapsed entity's label. */
             AST_GraphEntity *collapsed_entity = MatchClause_GetEntity(ast->matchNode, exp->operand.variadic.alias);
 
-            /* Failed to find collapsed entity. */
+            /* Invalid query, return clause refers to non-existent entity. */
             if(collapsed_entity == NULL) {
```

Comment:
Assert.
This shouldn't happen as we've passed the first AST validation.
asserting here would also remove the clean up logic you've presented.